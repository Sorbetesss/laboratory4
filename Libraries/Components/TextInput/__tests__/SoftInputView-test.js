/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @format
 * @emails oncall+react_native
 * @flow strict-local
 */

'use strict';

const React = require('react');
const View = require('../../View/View');
const SoftInputView = require('../SoftInputView');
const render = require('../../../../jest/renderer');

describe('<SoftInputView />', () => {
  it('should render as <RCTSoftInputView> when mocked', () => {
    const instance = render.create(
      <SoftInputView nativeID="1">
        <View />
      </SoftInputView>,
    );
    expect(instance).toMatchSnapshot();
  });

  it('should shallow render as <SoftInputView> when mocked', () => {
    const output = render.shallow(
      <SoftInputView nativeID="1">
        <View />
      </SoftInputView>,
    );
    expect(output).toMatchSnapshot();
  });

  it('should shallow render as <SoftInputView> when not mocked', () => {
    jest.dontMock('../SoftInputView');

    const output = render.shallow(
      <SoftInputView nativeID="1">
        <View />
      </SoftInputView>,
    );
    expect(output).toMatchSnapshot();
  });

  it('should render as <RCTSoftInputView> when not mocked', () => {
    jest.dontMock('../SoftInputView');

    const instance = render.create(
      <SoftInputView nativeID="1">
        <View />
      </SoftInputView>,
    );
    expect(instance).toMatchSnapshot();
  });
});
