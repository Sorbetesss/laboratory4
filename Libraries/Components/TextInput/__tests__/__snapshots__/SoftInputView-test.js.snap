// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SoftInputView /> should render as <RCTSoftInputView> when mocked 1`] = `
<RCTSoftInputView
  nativeID="1"
  style={
    Array [
      undefined,
      Object {
        "position": "absolute",
      },
    ]
  }
>
  <View />
</RCTSoftInputView>
`;

exports[`<SoftInputView /> should render as <RCTSoftInputView> when not mocked 1`] = `
<RCTSoftInputView
  nativeID="1"
  style={
    Array [
      undefined,
      Object {
        "position": "absolute",
      },
    ]
  }
>
  <View />
</RCTSoftInputView>
`;

exports[`<SoftInputView /> should shallow render as <SoftInputView> when mocked 1`] = `
<SoftInputView
  nativeID="1"
>
  <View />
</SoftInputView>
`;

exports[`<SoftInputView /> should shallow render as <SoftInputView> when not mocked 1`] = `
<SoftInputView
  nativeID="1"
>
  <View />
</SoftInputView>
`;
